### PR TITLE
Fix bug in parser + more tests

### PR DESCRIPTION
**Pull request by allanrenucci**

### CODE REVIEW

File: scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/XmlParser.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ class XmlParser(Block: P0,
 
     val Reference = P( EntityRef | CharRef )
     val EntityRef = P( "&" ~ Name ~/ ";" )
-    val CharRef   = P( "&#" ~ Num ~/ ";" | "&#x" ~ HexNum ~/ ";" )
```
Comment (olafurpg) [inline scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/XmlParser.scala:56]: Nice catch!

File: scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala [hunk 2]
```diff
@@ -974,7 +974,8 @@ class LegacyScanner(input: Input, dialect: Dialect) {
     val xmlParser = new XmlParser(embeddedScalaExprPositions)
     val result: Int = xmlParser.XmlExpr.parse(new String(input.chars), index = start) match {
       case Parsed.Success(_, endExclusive) => endExclusive - 1
-      case Parsed.Failure(_, _, _) => syntaxError("malformed xml literal", at = start)
+      case foo @ Parsed.Failure(expected, failIndex, _) =>
+        syntaxError(s"malformed xml literal, expected: $expected", at = failIndex)
```
Comment (olafurpg) [inline scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala:978]: Sweet 👍

**Review by olafurpg — APPROVED** [review]
Amazing work. It's a great relief to see more xml literal tests, this is one part of the scalameta parser that I've always been uneasy about.

**Comment by olafurpg** [comment]
I ran the parser property tests and got the same 46 errors, of which I listed a few in #1 

Could you possibly take a look at those? The links point to the culprit lines so it should be fairly simple to minimize some of those bugs.

**Comment by allanrenucci** [comment]
I was able to reproduce the bug. Still investigating.

**Pull request merged**